impress-org/givewp
# Feature: Initial react page setup

## Description

The changes in this PR include initial setup to support the new Donation detail page as a react application and scaffolds its basic frontend structure.

- The Donations resource directory was updated to include an additional react app. 
- The DonationsAdminPage class was refactored to render a new entry point for the app and to enqueue it's scripts.
- The installation of JOI and React hook form
- Updates to webpack to load the files.

## Affects
Donation Admin Page

## Testing Instructions

From the Donation Table page select the edit button on any donation row. The new react page should render with the text "donation-details"

## Pre-review Checklist
